tests/headingcommand.js

```diff
@@ -171,13 +158,13 @@ describe( 'HeadingCommand', () => {
 				convertTo = option;
 			}
 
-			it( 'converts to default option when executed with already applied option', () => {
+			it( 'does nothing when executed with already applied option', () => {
 				const command = commands.heading1;
 
 				setData( document, '<heading1>foo[]bar</heading1>' );
 				command.execute();
 
-				expect( getData( document ) ).to.equal( '<paragraph>foo[]bar</paragraph>' );
+				expect( getData( document ) ).to.equal( '<heading1>foo[]bar</heading1>' );
```

Comment:
This change doesn't check if it does nothing. It'd be good to listen to `change` event or something. Right now, if you remove the `if ( !block.is( this.modelElement ) )` condition the test still pass. And the thing is that we don't want the rename operation to be applied at all.